### ISSUE TITLE
[Python] Implement new Concept Exercise: string formatting

### ISSUE THREAD

**Issue by DavidGerva** [exercism/v3#1647]
This issue describes how to implement the `string formatting` concept exercise for the python track, explaining all the various methods.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

This concept exercise should convey a basic understanding of the various `string formatting` methods in python.

## Learning objectives

- Types of string formatting, and [when each is best used](https://realpython.com/python-string-formatting/#which-string-formatting-method-should-you-use) and [deferring i18n use cases for f-strings](https://www.python.org/dev/peps/pep-0501/#deferring-consideration-of-possible-use-in-i18n-use-cases).
- ["Old Style"](https://docs.python.org/3/library/stdtypes.html#printf-style-string-formatting) formatting with the `%` operator
- "New Style" formatting with [`str.format()`](https://docs.python.org/3/library/string.html#format-string-syntax)
-  String interpolation, or ["f-strings"](https://docs.python.org/3/reference/lexical_analysis.html#formatted-string-literals)
-  Standard Library [template strings](https://docs.python.org/3/library/string.html#template-strings) (_import statement required_)
- Format specification ["mini-language"](https://docs.python.org/3/library/string.html#format-specification-mini-language) e.g. formatting an int variable as a hexadecimal string, truncating long strings etc

## Out of scope

- Translation -- internationalization and localization of strings and string formats
- `str.maketrans` and [`str.translate`](https://docs.python.org/3/library/stdtypes.html#str.translate)
- `gettext`

## Concepts

- string formatting with the `%` operator
- new style string formatting with `str.format()`
- string interpolation via f-strings
- template strings

## Prerequisites

- `basics`
- `strings`

## Resources to refer to

- [str official doc](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str)
- [str methods](https://docs.python.org/3/library/stdtypes.html#string-methodsstrst)
- [Format String syntax (Python docs)](https://docs.python.org/3/library/string.html#format-string-syntax)
- ["printf" style string formatting (Python docs)](https://docs.python.org/3/library/stdtypes.html#old-string-formatting)
- [Formatted string literals (Python docs)](https://docs.python.org/3/reference/lexical_analysis.html#formatted-string-literals)
- [Format Specification Mini-Language](https://docs.python.org/3/library/string.html#formatspec)
- [Template Strings (Python docs)](https://docs.python.org/3/library/string.html#template-strings)
- [Format examples (Python docs)](https://docs.python.org/3/library/string.html#formatexamples)
- [Guide to Python string formatting](https://realpython.com/python-string-formatting/)
- [Which Method to Use?](https://realpython.com/python-string-formatting/#which-string-formatting-method-should-you-use)
- [Customizing string formatting (official docs)](https://docs.python.org/3/library/string.html#string-formatting)

### Hints

- refer to one or more of the documentation links above.

### After

Deeper discussions of pros and cons of each  method.  Security implications of string interpolation and user input.  Unicode, internationalization and localization considerations and organization. 

## Representer

No changes required.

## Analyzer

No changes required.

## Implementing

Tests should be written using unittest.TestCase and the test file named bool_basic_test.py.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

## Edits
-

**Comment by BethanyG** [comment]
@J08K  -- are you still working on this?  Would you like some help or collaboration?  @mohanrajanr has some time free to help and wanted to know if there is anything he could finish up or help you with?  If so, just ping him or myself in Slack.  😄

**Comment by BethanyG** [comment]
Note:  https://github.com/exercism/v3/pull/2204 is the Draft PR for this issue.

**Comment by J08K** [comment]
> @J08K  -- are you still working on this?  Would you like some help or collaboration?  @mohanrajanr has some time free to help and wanted to know if there is anything he could finish up or help you with?  If so, just ping him or myself in Slack.  😄 

I want to get back to this ASAP, I had a test week last week (A.K.A. A lot of very difficult tests in a week), but I am sure I can work on it again. When I do I'll see if @mohanrajanr can help! Thanks

**Comment by valentin-p** [comment]
I have created a PR with a concept exercise for this, hopefully, we can merge it before the first beta release of the v3.
